src/App.ts

```diff
@@ -30,17 +29,17 @@ export class App {
 
   listenToServerMessages(){
     this.client.on("message", async (message: Message) =>{
-      this.messageHandler.receivedMessage(message, this.client)
+      this.messageHandler.receivedMessage(message)
     });
   }
 
-  botLogin(){
-    this.client.login(config.BOT_TOKEN);
+  botLogin(botToken: string){
+    this.client.login(botToken);
     console.log("Bot is running.")
   }
 
-  constructor(){
-    this.botLogin()
+  constructor(botToken: string){
```

Comment:
nit: could we run all the setup in a run function rather than the ctor? That way in index.js we call run on the App object rather than just instantiating it.